File: odroid/Motor/src/bert/motor/controller/MotorManager.java
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 *                 MIT License.
 */
package bert.motor.controller;

import bert.share.message.MessageBottle;
/**
 *  This interface is satisfied by the MotorGroupController and describes
 *  the callbacks utilized by individual MotorContrtollers
 */
public interface MotorManager  {
	public void handleAggregatedResponse(MessageBottle response);
	public void handleSynthesizedResponse(MessageBottle response);
	public void handleSingleMotorResponse(MessageBottle response);
	public int getControllerCount();
}

File: Bert/motor/src/main/kotlin/chuckcoughlin/bert/motor/controller/MotorManager.kt
/**
 * Copyright 2022-2023. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.motor.controller

import chuckcoughlin.bert.common.message.MessageBottle
/**
 * This interface is satisfied by the MotorGroupController and describes
 * the callbacks utilized by individual MotorControllers
 */
interface MotorManager {
    suspend fun handleAggregatedResponse(cname:String,response: MessageBottle):MessageBottle
    suspend fun handleSynthesizedResponse(response: MessageBottle):MessageBottle
    suspend fun handleSingleControllerResponse(response: MessageBottle):MessageBottle

    var controllerCount: Int
}